### PR TITLE
rtmp-services: Add "wavezoo" RTMPS service

### PR DESCRIPTION
**Pull request by mariocoellocastro**
Wavezoo.com streaming server settings for DJ's

### Description
Adding 4 new types of server settings for with 10 servers each type

### Motivation and Context
Our DJ's need a simple way for configuring OBS properly

### How Has This Been Tested?
This was tested with the last release of OBS in a Linux environment (Ubuntu)

### Types of changes

New feature (non-breaking change which adds functionality): New Wavezoo specific stream settings and servers

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by derrod — CHANGES_REQUESTED** [review]
Please read the [Service Submission Guidelines](https://github.com/obsproject/obs-studio/wiki/Service-Submission-Guidelines).

You're also adding the service multiple times which is not necessary. On top of that, you have marked the service as common and not provided any valid ingest servers and the website listed does not appear to be available.

**Comment by mariocoellocastro** [comment]
You're right. I missed the guidelines.

I'll do the proper changes and try again when is ready.

Thanks and sorry for the inconvenience.

Mario

**Comment by Fenrirthviti** [comment]
You can update your branch and force push the changes, which is preferred rather than opening a new PR.

File: plugins/rtmp-services/data/services.json [hunk 1]
```diff
@@ -2846,6 +2846,262 @@
...
+            "supported video codecs": [
+                "h264"
+            ],
+            "supported audio codecs": [
+                "aac"
+            ],
+            "servers": [
+                {
+                    "name": "Server #01 (RTMP)",
+                    "url": "rtmp://live-stream-server-01:1935/LiveApp"
```
Comment (norihiro) [inline plugins/rtmp-services/data/services.json:2863]: Is this hostname (`live-stream-server-01`) correct? You might need to add your domain name, which is automatically added when you attempt to resolve the hostname.

**Comment by mariocoellocastro** [comment]
No, I'm really ashamed for that. I did nothing well. The next commit will
be fine.

File: plugins/rtmp-services/data/services.json [hunk 2]
```diff
@@ -2846,6 +2846,139 @@
...
+                }
+            ],
+            "protocol": "RTMPS",
+            "supported video codecs": [
+                "h264",
+                "hevc"
+            ],
+            "supported audio codecs": [
+                "aac",
+                "opus"
```
Comment (derrod) [inline plugins/rtmp-services/data/services.json:2943]: Unless your servers support [enhanced RTMP](https://github.com/veovera/enhanced-rtmp) do not specify "hevc" and "opus" here (or omit those properties entirely).

File: plugins/rtmp-services/data/services.json [hunk 3]
```diff
@@ -2846,6 +2846,139 @@
...
+                        "res": "852x480",
+                        "fps": 30,
+                        "max bitrate": 1500
+                    }
+                ],
+                "max fps": 60,
+                "max video bitrate": 14000,
+                "max audio bitrate": 192,
+                "bframes": 2,
+                "x264opts": "tune=zerolatency scenecut=0"
```
Comment (derrod) [inline plugins/rtmp-services/data/services.json:2980]: `tune=zerolatency` negatively impacts the quality more than it saves latency, especially if you're allowing b-frames you're better off not having it.

File: plugins/rtmp-services/data/services.json [hunk 4]
```diff
@@ -2846,6 +2846,139 @@
...
+        },
+        {
+            "name": "Wavezoo",
+            "common": false,
+            "more_info_link": "https://blog.wavezoo.com/obs-quality-guide/",
+            "stream_key_link": "https://oozevaw.com/user/bookings?filter=SETTLED_AND_ON_AIR",
+            "servers": [
+                {
+                    "name": "Live Stream Server #01",
+                    "url": "rtmps://live-stream-server-01.wavezoo.com:8443/LiveApp"
```
Comment (derrod) [inline plugins/rtmp-services/data/services.json:2858]: None of the RTMP URLs resolve.

**Review by derrod — CHANGES_REQUESTED** [review]

**Comment by derrod** [comment]
The RTMP URLs you specified still don't resolve to any IPs.

**Comment by mariocoellocastro** [comment]
Sure. It’s a pool of server names, the server is created on demand for an
specific event. Right now, there’s no server instance behind the URLs.

**Comment by norihiro** [comment]
If the server is not running, it's eligible to be removed like the PR 9387.

**Comment by mariocoellocastro** [comment]
Hi Nirihiro,

   As I explained before, we run the servers on demand, but the URLs are
fixed. If you want to try wavezoo.com I can give you a demo. When an event
(a party in our case) is about to start, we assign a server of the pool to
the party, and the server (a big and expensive one) starts. Then, the
streamer (DJ) connects to the assigned server using the provided stream ID
in order to authenticate the user.

As you can see, we cannot have on line 20 big and expensive servers 24/7.

I appreciate your understanding.

Kind regards,

    Mario

**Comment by Fenrirthviti** [comment]
After some discussion, I am making the call to decline this PR for now.

Wavezoo operates on a non-standard way and, frankly, is a bit of an anti-pattern to how every other streaming service operates with servers that are not expected to be available unless explicitly requested.

It conflicts with the user expectations, and would require custom handling to avoid our service validation checks from failing (as these servers would be offline more often than not). No other services in OBS really operate this way, so this is very much a non-standard way of operating and we'd be carving out a lot of exceptions I would rather not set a precedent for.

If there is an update towards a more standard service setup, we will happily re-review at that time. Thank you for your understanding.

**Pull request closed without merging**